registration/management/commands/import.py

```diff
@@ -0,0 +1,59 @@
+from django.db import transaction
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand, CommandError
+from participant.models import Troop
+
+import csv
+
+
+class Command(BaseCommand):
+    help = 'Imports a set of users from a CSV file'
+
+    def add_arguments(self, parser):
+        parser.add_argument('filename')
+
+    def handle(self, *args, **options):
+        created_users, created_troops = 0, 0
+
+        try:
+            with open(options['filename'], newline='') as csv_file:
+                csv_reader = csv.reader(csv_file, delimiter=';')
+                columns = self._extract_columns(next(csv_reader))
+
+                with transaction.atomic():
```

Comment:
I think that a line should be atomic (and not the whole file).

For instance when we start sending email to newly created users

If the second email fails (network issue or some other problem), then an exception will be thrown and nothing will be saved in the database : the first user receives an email that can't work (since it is not anymore in the database).
And we we re-import, the emails are sent again.

With an "atomicity per-line" (and get_or_create to ignore already created users) we will be much more robust:

```python
with transaction.atomic():
  user.get_or_create()
  send_email()
```
We can repeat the import operation many times and no double email will be sent :-)

Reply:
Hm, makes sense.